doc/enterprise/s3gateway.md

```diff
@@ -1,47 +1,551 @@
+to access Pachyderm. Complete the steps in one of the sections below that
+correspond to your S3 client.
+
+### Configure MinIO
+
+If you are using AWS CLI, skip this section.
+
+To install and configure MinIO, complete the following steps:
+
+1. Install the MinIO client on your platform as
+described on the [MinIO download page](https://min.io/download#/macos).
+1. Verify that MinIO components are successfully installed by running
+the following command:
+
+   ```bash
+   $ minio version
+   $ mc version
+   Version: 2019-07-11T19:31:28Z
+   Release-tag: RELEASE.2019-07-11T19-31-28Z
+   Commit-id: 31e5ac02bdbdbaf20a87683925041f406307cfb9
+   ```
+
+1. Set up the MinIO configuration file to use the `30600` port for your host:
+
+   ```bash
+   vi ~/.mc/config.json
+   ```
+
+   You should see a configuration similar to the following:
+
+   * For a minikube deployment, verify the
+   `local` host configuration:
+
+     ```bash
+     "local": {
+               "url": "http://localhost:30600",
+               "accessKey": "",
+               "secretKey": "",
+               "api": "S3v4",
+               "lookup": "auto"
+           },
+     ```
+
+### Configure the AWS CLI
+
+If you are using the MinIO client, skip this section.
+
+If you have not done so already, you need to install and
+configure the AWS CLI client on your machine. You need to
+provide the AWS Access Key ID and the AWS Secret Access Keys
+for the account that has access to the S3 bucket that you want
+to use with Pachyderm.
+To configure the AWS CLI, complete the following steps:
+
+1. Install the AWS CLI for your operating system as described
+in the [AWS documentation](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-install.html).
+
+1. Verify that the AWS CLI is installed:
+
+   ``bash
+   $ aws --version aws-cli/1.16.204 Python/2.7.16 Darwin/17.7.0 botocore/1.12.194
+   ```
+
+1. Configure AWS CLI:
+
+   ```bash
+   $ aws configure
+   AWS Access Key ID:
+   AWS Secret Access Key:
+   Default region name:
+   Default output format [None]:
+   ```
+
+### Configure `s3cmd`
+
+S3cmd is an open-source command line client that enables you
+to access S3 object store buckets. To configure s3cmd, complete
+the following steps:
+
+1. If you do not have S3cmd installed on your machine, install
+it as described in the [S3cmd documentation](https://s3tools.org/download).
+For, example, in macOS, run:
+
+   ```bash
+   $ brew install s3cmd
+   ```
+
+1. Verify that s3cmd is installed:
+
+   ```bash
+   $ s3cmd --version
+   s3cmd version 2.0.2
+   ```
+
+1. Configure S3cmd to use Pachyderm:
+
+   ```bash
+   $ s3cmd --configure
+     ...
+   ```
+
+1. Fill all fields and specify the following settings for Pachyderm.
+
+   **Example:**
+
+   ```bash
+
+   New settings:
+     Access Key: ""
+     Secret Key: ""
+     Default Region: US
+     S3 Endpoint: localhost:30600
+     DNS-style bucket+hostname:port template for accessing a bucket: 30600/%(bucket)
```

Comment:
DNS-style bucket+hostname:port template for accessing a bucket: localhost:30600/%(bucket)
Write

Reply:
Somehow dropped the "localhost" here...